src/plugin/ControlBar/AnimationProgressBar.ts

```diff
@@ -0,0 +1,334 @@
+  private _firstTouch: { x: number; y: number } | null;
+  private _scrolling: boolean;
+  private _origTimeScale: number;
+
+  /** */
+  public constructor(view3D: View3D, controlBar: ControlBar, {
+    position = ControlBar.POSITION.TOP,
+    order = 9999
+  }: Partial<AnimationProgressBarOptions> = {}) {
+    this.position = position;
+    this.order = order;
+
+    this._view3D = view3D;
+    this._controlBar = controlBar;
+    this._createElements();
+    this._enabled = false;
+    this._firstTouch = null;
+    this._scrolling = false;
+    this._origTimeScale = 1;
+  }
+
+  /**
+   * Enable control item
+   */
+  public enable() {
+    const view3D = this._view3D;
+    if (this._enabled) return;
+
+    this._onResize();
+    this._enabled = true;
+
+    view3D.on(EVENTS.RENDER, this._onRender);
+    view3D.on(EVENTS.RESIZE, this._onResize);
+    view3D.on(EVENTS.MODEL_CHANGE, this._onModelChange);
+
+    this.enableInput();
+  }
+
+  /**
+   * Disable control item
+   */
+  public disable() {
+    const view3D = this._view3D;
+    if (!this._enabled) return;
+
+    this._enabled = false;
+
+    view3D.off(EVENTS.RENDER, this._onRender);
+    view3D.off(EVENTS.RESIZE, this._onResize);
+    view3D.off(EVENTS.MODEL_CHANGE, this._onModelChange);
+
+    this.disableInput();
+  }
+
+  /**
+   * Enable mouse / touch inputs
+   */
+  public enableInput() {
+    const root = this._rootEl;
+    const view3D = this._view3D;
+
+    this._firstTouch = null;
+    this._scrolling = false;
+
+    if (view3D.animator.animationCount <= 0) return;
+
+    root.addEventListener(BROWSER.EVENTS.MOUSE_DOWN, this._onMouseDown);
+
+    root.addEventListener(BROWSER.EVENTS.TOUCH_START, this._onTouchStart, { passive: false });
+    root.addEventListener(BROWSER.EVENTS.TOUCH_MOVE, this._onTouchMove, { passive: false });
+    root.addEventListener(BROWSER.EVENTS.TOUCH_END, this._onTouchEnd);
+  }
+
+  /**
+   * Disable mouse / touch inputs
+   */
+  public disableInput() {
+    const root = this._rootEl;
+
+    root.removeEventListener(BROWSER.EVENTS.MOUSE_DOWN, this._onMouseDown);
+    window.removeEventListener(BROWSER.EVENTS.MOUSE_MOVE, this._onMouseMove, false);
+    window.removeEventListener(BROWSER.EVENTS.MOUSE_UP, this._onMouseUp, false);
+
+    root.removeEventListener(BROWSER.EVENTS.TOUCH_START, this._onTouchStart);
+    root.removeEventListener(BROWSER.EVENTS.TOUCH_MOVE, this._onTouchMove);
+    root.removeEventListener(BROWSER.EVENTS.TOUCH_END, this._onTouchEnd);
+  }
+
+  private _createElements() {
+    const controlBar = this._controlBar;
+    const className = {
+      ...controlBar.className,
+      ...ControlBar.DEFAULT_CLASS
+    };
+
+    const root = document.createElement(BROWSER.EL_DIV);
+    root.classList.add(className.PROGRESS_ROOT);
+    root.draggable = false;
+
+    const track = document.createElement(BROWSER.EL_DIV);
+    track.classList.add(className.PROGRESS_TRACK);
+
+    const knob = document.createElement(BROWSER.EL_DIV);
```

Comment:
Maybe the knob is a thumb?

Sliders and video players seem to use the word "thumb" more often.